catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/WhenNfcBrickTest.java

```diff
@@ -0,0 +1,488 @@
+
+		pressBack();
+
+		spinnerValuesResourceIdsContained = Arrays.asList(
+				getResourcesString(R.string.brick_when_nfc_default_all),
+				getResourcesString(R.string.new_nfc_tag),
+				getResourcesString(R.string.test_tag_name_1),
+				getResourcesString(R.string.test_tag_name_2),
+				getResourcesString(R.string.default_tag_name),
+				getResourcesString(R.string.default_tag_name) + "_1");
+		spinnerValuesResourceIdsNotContained = Arrays.asList();
+		checkNfcSpinnerContains(nfcBrickPosition, spinnerValuesResourceIdsContained, spinnerValuesResourceIdsNotContained);
+
+		onScriptList().atPosition(nfcBrickPosition).onChildView(withId(R.id.brick_when_nfc_spinner))
+				.perform(click());
+		onView(withText(R.string.default_tag_name))
+				.check(matches(isDisplayed()));
+	}
+
+	@Test
+	public void testNfcSensorVariable() throws InterpretationException {
+		NdefMessage ndefMessage1 = NfcHandler.createMessage(NFC_NDEF_STRING_1, BrickValues.TNF_MIME_MEDIA);
+		NdefMessage ndefMessage2 = NfcHandler.createMessage(NFC_NDEF_STRING_2, BrickValues.TNF_MIME_MEDIA);
+
+		UiHardwareTestUtils.gotoNfcFragment(nfcBrickPosition);
+		UiHardwareTestUtils.fakeNfcTag(FIRST_TEST_TAG_ID, ndefMessage1, null);
+
+		onView(withId(R.id.button_play))
+				.perform(click());
+
+		Assert.assertTrue("Timeout while waiting for user variable change - 'tag id'",
+				userVariableEqualsWithinTimeout(readTagId, 0, 2000));
+		Assert.assertTrue("Timeout while waiting for user variable change - 'tag message' ",
+				userVariableEqualsWithinTimeout(readTagMessage, "0.0", 2000));
+
+		UiHardwareTestUtils.fakeNfcTag(SECOND_TEST_TAG_ID, ndefMessage2, null);
+		Assert.assertTrue("Timeout while waiting for user variable change - 'tag id'",
+				userVariableEqualsWithinTimeout(readTagId, Double.parseDouble(TAG_ID_2), 2000));
+		Assert.assertTrue("Timeout while waiting for user variable change - 'tag message' ",
+				userVariableEqualsWithinTimeout(readTagMessage, NFC_NDEF_STRING_2, 2000));
+	}
+
+	@Test
+	public void testSelectTagAndPlay() {
+		checkIfSpinnerOnBrickAtPositionShowsString(R.id.brick_when_nfc_spinner, nfcBrickPosition,
+				R.string.brick_when_nfc_default_all);
+		clickSpinnerValueOnBrick(R.id.brick_when_nfc_spinner, nfcBrickPosition, R.string.test_tag_name_1);
+
+		onView(withId(R.id.button_play))
+				.perform(click());
+
+		Script script = ProjectManager.getInstance().getCurrentScene().getSpriteList().get(0).getScript(0);
+		assertEquals("tag not set", ((WhenNfcBrick) script.getScriptBrick()).getNfcTag().getNfcTagName(),
+				tagDataList.get(0).getNfcTagName());
+		pressBack();
+		pressBack();
+		checkIfSpinnerOnBrickAtPositionShowsString(R.id.brick_when_nfc_spinner, nfcBrickPosition,
+				R.string.test_tag_name_1);
+		clickSpinnerValueOnBrick(R.id.brick_when_nfc_spinner, nfcBrickPosition, R.string.test_tag_name_2);
+
+		onView(withId(R.id.button_play))
+				.perform(click());
+
+		script = ProjectManager.getInstance().getCurrentScene().getSpriteList().get(0).getScript(0);
```

Comment:
since you get a reference to the script, why do you call it again?
also, probably better to already save a reference to a script when you initialize it in the createProject instead of digging for it.